pxr/base/tf/stringUtils.h

```diff
@@ -683,18 +727,113 @@ std::string TfStringCatPaths( const std::string &prefix,
 inline bool
 TfIsValidIdentifier(std::string const &identifier)
 {
-    char const *p = identifier.c_str();
-    auto letter = [](unsigned c) { return ((c-'A') < 26) || ((c-'a') < 26); };
-    auto number = [](unsigned c) { return (c-'0') < 10; };
-    auto under = [](unsigned c) { return c == '_'; };
-    unsigned x = *p;
-    if (!x || number(x)) {
-        return false;
+    if (UseUTF8Identifiers())
+    {
+        // use unicode utils to validate the identifier
+        return TfUnicodeUtils::IsValidUTF8Identifier(identifier, 
+            identifier.begin(), identifier.end());
+    }
+    else
```

Comment:
Minor nit (and I defer to others on style), but it would be nice to not have the `else` statement here since the `if` has an early return.

Has a few benefits:
* Smaller diff since the insertion of the new branch would be the only thing that changed.
* Easier to catch if there were any unintended changes in the existing code (since the diff highlighting doesn't delineate between indented code and modified code very well on Github)
* Reduces code indentation which (IMHO) is nice for readability

Reply:
I think these are reasonable points, and once the PR has been approved in principle this seems like a change worth having.  The reason it is structured this way at the moment is so that it is abundantly clear to the reviewer that the original code paths are intact in almost all instances.  Once of the biggest concerns with a change like this is "will it introduce subtle errors or differences into my applications / workflows".  For example, in a first pass I had modified the original lexer / parser to support both the original and new workflows, but it had the potential to introduce subtle errors because the new lexer was more permissive than the old and relied on additional validation methods to be sure what we lexed was the right token type.  I had hoped that the way it is structured now makes it much clearer that the original code paths are untouched, with the additional overhead of the runtime check for the new switch.  The only place where I diverged from this philosophy was the introduction of separate validation rules for prim names, where the child policy on prims was changed to use the  new prim name validation rules (albeit when the switch is off it actually falls back to the same behavior, but it's not necessarily obvious to the reviewer).